content/blog/2023-12-01-task-lifecycle.md

```diff
@@ -0,0 +1,572 @@
+Title: Lifecycle of a kernel task
+Date: 2023-12-01 08:00:00
+Authors: Benjamin Gray
+Category: Development
+Tags: linux
+
+## Introduction
+
+CPU cores are limited in number. Right now my computer tells me it's running
+around 500 processes, and I definitely do not have that many cores. The
+operating systems ability to virtualise work as independent 'executable units'
+and distribute them across the limited CPU pool is one of the foundations of
+modern computing.
+
+The Linux kernel calls these virtual execution units _tasks_. Each task
+encapsulates all the information the kernel needs to swap it in and out of
+running on a CPU core. This includes register state, memory mappings, open
+files, and any other resource that needs to be tied to a particular task. Nearly
+every workload in the kernel, including kernel background jobs and userspace
```

Comment:
I changed it to "Nearly every work item in the kernel, ..., is"

There are cases where you don't run in a task context (log lines will use `C` and the core number instead of `T` and the task number).